packages/odyssey-react-mui/src/labs/DataFilters.tsx

```diff
@@ -410,22 +464,26 @@ const DataFilters = ({
                               ] as string) ?? ""
                             }
                             onChange={(ev) =>
-                              handleInputChange(
-                                filterPopoverCurrentFilter.id,
-                                ev.currentTarget.value
-                              )
+                              updateInputValue({
+                                filterId: filterPopoverCurrentFilter.id,
+                                value: ev.currentTarget.value,
+                              })
                             }
                             endAdornment={
                               inputValues[filterPopoverCurrentFilter.id] && (
                                 <MuiIconButton
                                   size="small"
                                   aria-label="Clear filter"
                                   onClick={() => {
-                                    handleInputChange(
-                                      filterPopoverCurrentFilter.id,
-                                      undefined,
-                                      true
-                                    );
+                                    updateInputValue({
+                                      filterId: filterPopoverCurrentFilter.id,
+                                      value: undefined,
+                                    });
+
+                                    updateFilters({
```

Comment:
Kevin and I also discussed splitting the logic for updating the filter value shown in the  DataFilter UI and the actual value in the form components into 2 parts ` updateInputValue` and `updateFilters` to separate concerns and make the code a little easier to read.